sdk/monitor/azure-monitor-opentelemetry-exporter/tests/quickpulse/test_utils.py

```diff
@@ -425,3 +456,186 @@ def test_get_metrics_from_projections(self, projection_map_mock):
                 ("test-id4", 4.0),
             ],
         )
+
+    def test_validate_info_invalid_telemetry_type(self):
+        valid_metric_info = mock.Mock(telemetry_type="Request", projection=None, filter_groups=[])
+        valid_metric_info.telemetry_type = "INVALID_TYPE"
+        self.assertFalse(_validate_derived_metric_info(valid_metric_info))
+
+    def test_validate_info_custom_metric_projection(self):
+        valid_metric_info = mock.Mock(telemetry_type="Request", projection=None, filter_groups=[])
+        valid_metric_info.telemetry_type = "Request"
+        valid_metric_info.projection = "CustomMetrics.MyMetric"
+        self.assertFalse(_validate_derived_metric_info(valid_metric_info))
+
+    @mock.patch("azure.monitor.opentelemetry.exporter._quickpulse._utils._validate_filter_predicate_and_comparand")
+    @mock.patch("azure.monitor.opentelemetry.exporter._quickpulse._utils._validate_filter_field_name")
+    def test_validate_info_valid_filters(self, field_mock, pred_mock):
+        valid_metric_info = mock.Mock(telemetry_type="Request", projection=None, filter_groups=[])
+        filter = mock.Mock()
```

Comment:
I'm not too familiar with the python mocking framework - where is the filter mock actually defined? Ie how do I know what it is filtering for